### PR TITLE
Declare key/value format for PRODUCER in kafka file

### LINKED ISSUE #113: Declare key/value format for PRODUCER in kafka file
Improve kafka file PRODUCER (when https://github.com/jlandersen/vscode-kafka/pull/98 will be merged)  to declare key/value serializer:

 * `key-format` = none, long, integer, etc
 * `value-format` = none, long, integer, etc

Example:

```
PRODUCER
topic: prices
value-format: long
```

In the future we could add avro schema.

### PR DESCRIPTION
**Pull request by angelozerr**
Declare key/value format for PRODUCER in kafka file

Fixes #113

Signed-off-by: azerr azerr@redhat.com

### CODE REVIEW

**Comment by angelozerr** [comment]
I need to write tests and doc but it should start to work.

**Comment by angelozerr** [comment]
The PR can be reviewed now. There are tests and docs.

File: docs/Producing.md [hunk 1]
```diff
@@ -23,6 +23,26 @@ To produce a single record, click on the `Produce record` link above the `PRODUC
...
 The log about produced messages is printed in the `Kafka Producer Log` Output view.
 
+The `PRODUCER` block defines:
+
+ * `keyed message` which is declared after PRODUCER *[optional]*.
+ * `key`: the key *[optional]*.
+ * `key-format` : [serializer](#Serializer) to use for the key *[optional]*.
+ * `value-format` : [serializer](#Serializer) to use for the value *[optional]*.
+ 
+ * the rest of the content is the value upon `###` line is found.
```
Comment (fbricon) [inline docs/Producing.md:33]: until `###`

File: docs/Producing.md [hunk 2]
```diff
@@ -23,6 +23,26 @@ To produce a single record, click on the `Produce record` link above the `PRODUC
...
+
+### Serializer
+
+Serializer can have the following value:
```
Comment (fbricon) [inline docs/Producing.md:37]: The serializers

**Review by fbricon — CHANGES_REQUESTED** [review]

File: docs/Producing.md (hunk 1 above)
Comment (angelozerr) [reply docs/Producing.md]: fixed

File: docs/Producing.md (hunk 2 above)
Comment (angelozerr) [reply docs/Producing.md]: fixed

File: snippets/producers.json [hunk 3]
```diff
@@ -28,26 +29,58 @@
...
-		"description": "A producer generating keyed JSON records"
+        ],
+        "description": "A producer generating keyed JSON records"
+    },
+    "key-format-producer": {
+        "prefix": [
+            "producer"
+        ],
+        "body": [
+            "PRODUCER ${1:keyed-message}",
```
Comment (fbricon) [inline snippets/producers.json:40]: ${1:key-formatted-message}

File: snippets/producers.json [hunk 4]
```diff
@@ -28,26 +29,58 @@
...
+            ""
+        ],
+        "description": "A producer generating formatted keyed records"
+    },
+    "value-format-producer": {
+        "prefix": [
+            "producer"
+        ],
+        "body": [
+            "PRODUCER ${1:keyed-message}",
```
Comment (fbricon) [inline snippets/producers.json:56]: ${1:formatted-message}

**Review by fbricon — CHANGES_REQUESTED** [review]

File: snippets/producers.json (hunk 3 above)
Comment (angelozerr) [reply snippets/producers.json]: fixed

File: snippets/producers.json (hunk 4 above)
Comment (angelozerr) [reply snippets/producers.json]: fixed

**Comment by fbricon** [comment]
Thanks @angelozerr!

**Pull request merged**